### ISSUE TITLE
search_on an array field throws undefined method error

### ISSUE THREAD

**Issue by monfresh** [joshuabowers/search_magic#37]
Hi,

I just found out about this gem, and it looks promising, but I've come across an issue with searching on an array field. I have 3 embedded models:
Organization embeds_many :programs
Program embeds_many :locations
In the Location model, I have `field :keywords, type: Array` and `search_on :keywords`. When I try to seed the DB via a JSON file that has the same structure, it aborts with `undefined method 'keywords' for #<Array:0x007fbb2908ac50>`. If I remove `search_on :keywords`, I am able to run the rake task.

Why is `search_magic` choking on array fields?

I tried both the Rubygems version and the experimental branch, and they're both throwing that error.

**Comment by monfresh** [comment]
It looks like it's not just Array fields. Any field in the Location model I search_on throws the same error during the rake task. It also happens if the Models use has_many and belongs_to instead of embedded relationships.

**Comment by joshuabowers** [comment]
Hello monfresh,

As you have given the experimental branch a whirl, would you mind testing the latest commit against your use case, if you are still interested in using the gem? I'm fairly certain I have a fix in for the problem you were encountering, but it would be nice to test it against something a bit more real than a hastily built testing scenario.

For context to the problem you were experiencing: including the SearchMagic module adds a before_save callback on each document, which is responsible for bundling the appropriate data together for later searches. That callback, in turn, ends up navigating the search graph generated when you specify which data you want to search_on. (So, obviously enough, if you choose to not search on anything, the callback effectively does nothing.) The problem, near as I can tell, was that I was not properly handling a use case where a top level document would search through a many-to-many association with another document which, in itself, searches through a many-to-many. 

TL:DR: I needed to flatten some data I was passing up an inject stack.

**Comment by monfresh** [comment]
Hey @joshuabowers, thanks for your response, and sorry for responding so late. I ended up going with ElasticSearch via the Tire gem, so I haven't had a chance to play more with search_magic. If I ever do, I'll be sure to let you know.
